gateway-ha/src/test/java/io/trino/gateway/ha/router/TestRoutingGroupSelector.java

```diff
@@ -66,23 +90,153 @@ public void testByRoutingGroupHeader()
+                RoutingGroupSelector.byRoutingRulesEngine("src/test/resources/rules/routing_rules_trino_query_properties.yml", requestAnalyzerConfig);
+        HttpServletRequest mockRequest = prepareMockRequest();
+
+        when(mockRequest.getHeader(TrinoQueryProperties.TRINO_CATALOG_HEADER_NAME)).thenReturn("other_catalog");
+        when(mockRequest.getHeader(TrinoQueryProperties.TRINO_SCHEMA_HEADER_NAME)).thenReturn("other_schema");
+
+        assertThat(routingGroupSelector.findRoutingGroup(mockRequest)).isEqualTo("defaults-group");
+    }
+
+    @Test
+    void testTrinoQueryPropertiesQueryType()
+            throws IOException
+    {
+        RoutingGroupSelector routingGroupSelector =
+                RoutingGroupSelector.byRoutingRulesEngine("src/test/resources/rules/routing_rules_trino_query_properties.yml", requestAnalyzerConfig);
+        String query = "INSERT INTO foo SELECT 1";
+        Reader reader = new StringReader(query);
+        BufferedReader bufferedReader = new BufferedReader(reader);
+        HttpServletRequest mockRequest = prepareMockRequest();
+        when(mockRequest.getReader()).thenReturn(bufferedReader);
+
+        assertThat(routingGroupSelector.findRoutingGroup(mockRequest)).isEqualTo("type-group");
+    }
+
+    @Test
+    void testTrinoQueryPropertiesAlternateStatementFormat()
+            throws IOException
+    {
+        requestAnalyzerConfig.setClientsUseV2Format(true);
+        RoutingGroupSelector routingGroupSelector =
+                RoutingGroupSelector.byRoutingRulesEngine("src/test/resources/rules/routing_rules_trino_query_properties.yml", requestAnalyzerConfig);
+        String body = "{\"preparedStatements\" : {\"statement1\":\"INSERT INTO foo SELECT 1\"}, \"query\": \"EXECUTE statement1\"}";
+        Reader reader = new StringReader(body);
+        BufferedReader bufferedReader = new BufferedReader(reader);
+        HttpServletRequest mockRequest = prepareMockRequest();
+        when(mockRequest.getReader()).thenReturn(bufferedReader);
+
+        assertThat(routingGroupSelector.findRoutingGroup(mockRequest)).isEqualTo("type-group");
+    }
+
+    @Test
+    void testTrinoQueryPropertiesPreparedStatementInHeader()
+            throws IOException
+    {
+        String encodedStatements = "statement1=SELECT+%27s1%27+c%0A%0A,statement2=SELECT+%27s2%27+c%0A%0A,statement3=SELECT%0A++%27%2C%27+comma%0A%2C+%27%3D%27+eq%0A%0A,statement4=SELECT%0A++c1%0A%2C+c2%0AFROM%0A++foo%0A";
+
+        Enumeration<String> encodedStatementsEnumeration = new Enumeration<>()
+        {
+            int index;
+            final String[] elements = encodedStatements.split(",");
+
+            @Override
+            public boolean hasMoreElements()
+            {
+                return index < elements.length;
+            }
+
+            @Override
+            public String nextElement()
+            {
+                return elements[index++];
+            }
+        };
```

Comment:
Can be done by `Collections.enumeration(Arrays.asList(encodedStatements.split(",")))`